x-pack/plugins/security_solution/server/lib/detection_engine/signals/filter_events_with_list.ts

```diff
@@ -51,87 +96,97 @@ export const filterEventsAgainstList = async ({
     );
 
     if (exceptionItemsWithLargeValueLists.length === 0) {
-      logger.debug(buildRuleMessage('about to return original search result'));
+      logger.debug(
+        buildRuleMessage('no exception items of type list found - returning original search result')
+      );
       return eventSearchResult;
     }
 
-    // narrow unioned type to be single
-    const isStringableType = (val: SearchTypes) =>
-      ['string', 'number', 'boolean'].includes(typeof val);
-    // grab the signals with values found in the given exception lists.
-    const filteredHitsPromises = exceptionItemsWithLargeValueLists.map(
-      async (exceptionItem: ExceptionListItemSchema) => {
-        const { entries } = exceptionItem;
-
-        const filteredHitsEntries = entries
-          .filter((t): t is EntryList => entriesList.is(t))
-          .map(async (entry) => {
+    const valueListExceptionItems = exceptionsList.filter((listItem: ExceptionListItemSchema) => {
+      return listItem.entries.some((entry) => entriesList.is(entry));
```

Comment:
probably want `listItem.entries.every` here since we can't support items that have both list-based and regular entries

Reply:
We should also modify the exception list item schema to prevent creation of an item with both list and non-list entries, but that can be a separate PR.